Wikipedia/Code/ArticleToolbarController.swift

```diff
@@ -71,6 +76,56 @@ class ArticleToolbarController: Themeable {
         item.apply(theme: theme)
         return item
     }()
+    
+    lazy var moreButton: IconBarButtonItem = {
+        return createMoreButton()
+    }()
+    
+    private func createMoreButton(needsWatchButton: Bool = false, needsUnwatchHalfButton: Bool = false, needsUnwatchFullButton: Bool = false) -> IconBarButtonItem {
```

Comment:
Not suggesting we change this here, but wanted to mention something for consideration in the future.

```swift 
private func createMoreButton(needsWatchButton: Bool = false, needsUnwatchHalfButton: Bool = false, needsUnwatchFullButton: Bool)
```

I think this mutually exclusive boolean pattern is one of those internal API choices that potentially seeps unexpected behavioral expectations to callers down the line. I know in the actual logic there's some mutual exclusivity to do the right thing, but that doesn't prevent outside callers from using it in an unexpected way and getting unexpected (to them) results. For example, a caller can `createMoreButton(needsWatchButton: true, needsUnwatchHalfButton: true, needsUnwatchFullButton: true)`.

Though it's a little more lift (especially for minor buttons or whatever), I generally think wrapping these kinds of things in an `enum` is better API design: dramatically cleans up the function parameters and any call sites and makes it painfully obvious the only valid button configuration possible. Something like:

```swift

func createMoreButton(watchButtonConfiguration: WatchButtonConfiguration) { ... }

enum WatchButtonConfiguration {
	case watch
	case unwatchHalfFilled
	case unwatchFilled
	
	var image: UIImage {
		switch self {
		case .watch:
			return UIImage(systemName: "star") 
		case .unwatchHalfFilled:
			return UIImage(systemName: "star.leadinghalf.filled") 
		case .unwatchFilled:
			return UIImage(systemName: "star.fill") 		
		}
	}
}

moreButton = createMoreButton(watchButtonConfiguration: .unwatchHalfFilled)
```

Reply:
Agh I agree, sorry about that! I went ahead and made this change, but didn't push it in case you weren't keen on looking at it again at this time. I could put up a followup PR if you'd like.

Reply:
@tonisevener Oh cool – no worries. Yeah if you don't mind follow-up PR'ing it, I'd like to go ahead and merge this.